src/Sylius/Bundle/CoreBundle/OAuth/UserProvider.php

```diff
@@ -69,7 +69,7 @@ public function loadUserByOAuthUserResponse(UserResponseInterface $response): Us
             return $this->createUserByOAuthUserResponse($response);
         }
 
-        throw new UsernameNotFoundException('Email is null or not provided');
+        throw new UserNotFoundException('Email is null or not provided');
```

Comment:
Sorry if I repeat myself, but we may have a BC break here.

This polyfill is not enough as there is no way to know which exception is expected in a user-land `catch`.

Instead we could have a constructor arg (like a boolean flag) defined in a compiler pass that would check whether or not the new authentication system is wired (with `enable_authenticator_manager` set to `true` in the project configuration), it could be done by checking the existence of a definition of a service from the new layer.

Reply:
IMHO, there is nothing with new authenticator system, that was not thrown by Symfony here.
Before, we threw UsernameNotFoundAction which is removed so, we try to add a bc-layer to keep the catch working for an end user if his project is on Symfony 5.4. But if he upgrades on SF6, he must change his catch.
@lchrusciel do you remember our discussions about this deprecation?